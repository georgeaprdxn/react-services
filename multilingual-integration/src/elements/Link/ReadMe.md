Link

```jsx inside Markdown
import mockData from './__test__/Link.mock.json'
;<Link url={mockData.url} target={mockData.target} label={mockData.label} />
```
