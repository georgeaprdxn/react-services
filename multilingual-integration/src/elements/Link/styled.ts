import styled from 'styled-components'

const LinkElement = styled.a``

export { LinkElement }
