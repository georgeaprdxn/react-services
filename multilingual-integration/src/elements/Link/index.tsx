import React from 'react'
import { LinkElement } from './styled'

type LinkProps = {
	/**
	 * url to the href attribute
	 */
	url: string
	/**
	 * target preference
	 */
	target?: string
	/**
	 * label to the link
	 */
	label: string
} & typeof defaultProps

const LinkComp = (props: LinkProps) => {
	const { url, target, label } = props

	return (
		<LinkElement
			href={url}
			target={target}
			title={label}
			className="custom-link"
		>
			{label}
		</LinkElement>
	)
}

const defaultProps = {
	url: '#FIXME',
	target: '_self',
	label: '',
}

LinkComp.defaultProps = defaultProps

export default LinkComp
