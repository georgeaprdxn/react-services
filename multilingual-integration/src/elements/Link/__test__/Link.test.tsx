import React from 'react'
import { shallow } from 'enzyme'

import Link from '../index'
import mockLinkData from './Link.mock.json'

describe('Link element tests', () => {
	let linkElem: any = null
	beforeEach(() => {
		linkElem = shallow(
			<Link
				url={mockLinkData.url}
				target={mockLinkData.target}
				label={mockLinkData.label}
			/>
		)
	})

	it('should render a <a />', () => {
		expect(linkElem.find('.custom-link').length).toEqual(1)
	})

	it('validate inner text', () => {
		expect(linkElem.find('.custom-link').text()).toEqual(mockLinkData.label)
	})

	it('validates all props', () => {
		expect(linkElem.find('.custom-link').prop('href')).toEqual(mockLinkData.url)
		expect(linkElem.find('.custom-link').prop('title')).toEqual(
			mockLinkData.label
		)
		expect(linkElem.find('.custom-link').prop('target')).toEqual(
			mockLinkData.target
		)
	})

	it('matches the snaphot', () => {
		expect(linkElem).toMatchSnapshot()
	})
})
