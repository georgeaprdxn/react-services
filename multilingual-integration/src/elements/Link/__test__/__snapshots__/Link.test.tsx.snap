// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link element tests matches the snaphot 1`] = `
<styled.a
  className="custom-link"
  href="https://www.google.com"
  target="_blank"
  title="Google.com"
>
  Google.com
</styled.a>
`;
